src/app/data-access/services/data-access.service.ts

```diff
@@ -0,0 +1,180 @@
+import { AccessRequest, AccessRequestDialogData } from '../models/access-requests';
+
+/**
+ *  This service handles state and management of access requests (to datasets)
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class DataAccessService {
+  #dialogRef: MatDialogRef<DataAccessRequestModalComponent> | undefined;
+  #http = inject(HttpClient);
+  #auth = inject(AuthService);
+  #notification = inject(NotificationService);
+  #dialog = inject(MatDialog);
+  #userId = computed<string | null>(() => this.#auth.user()?.id || null);
+  #config = inject(ConfigService);
+  #arsBaseUrl = this.#config.arsUrl;
+  #arsEndpointUrl = `${this.#arsBaseUrl}/access-requests`;
+
+  showNewAccessRequestDialog = (datasetID: string) => {
+    const user = this.#auth.user();
+
+    if (!this.#auth.isAuthenticated() || !user) {
+      this.#notification.showError('You must be logged in to perform this action');
+      return;
+    }
+
+    const data: AccessRequestDialogData = {
+      datasetID,
+      email: user.email,
+      description: '',
+      fromDate: undefined,
+      untilDate: undefined,
+      isCanceled: false,
+      userId: '',
+    };
+
+    this.#dialogRef = this.#dialog.open(DataAccessRequestModalComponent, {
+      data,
+    });
+
+    this.#dialogRef
+      .afterClosed()
+      .subscribe((componentData) => this.#processAccessRequest(componentData));
+  };
+
+  #processAccessRequest = (data: AccessRequestDialogData) => {
+    const userid = this.#auth.user()?.id;
+    if (data.isCanceled || !this.#auth.isAuthenticated() || !userid) {
+      return;
+    }
+    data.userId = userid;
+    this.#performAccessRequest(data);
+  };
+
+  #performAccessRequest = (data: AccessRequestDialogData) => {
+    data.fromDate?.setUTCHours(0);
+    data.fromDate?.setUTCMinutes(0);
+    data.fromDate?.setUTCSeconds(0);
+    data.fromDate?.setUTCMilliseconds(0);
+    data.untilDate?.setHours(23);
+    data.untilDate?.setMinutes(59);
+    data.untilDate?.setSeconds(59);
+    data.untilDate?.setMilliseconds(999);
+    try {
+      firstValueFrom(
+        this.#http
+          .post<void>(this.#arsEndpointUrl, {
+            user_id: data.userId,
+            dataset_id: data.datasetID,
+            email: data.email,
+            request_text: data.description,
+            access_starts: data.fromDate?.toDateString(),
+            access_ends: data.untilDate?.toDateString(),
+          })
+          .pipe(
+            map(console.log),
+            catchError(async () =>
+              this.#notification.showError(
+                'There was an error submitting your access request.',
+              ),
+            ),
+          ),
+      ).then(async () => {
+        this.#accessRequests.reload();
+        this.#notification.showSuccess('Your request has been submitted successfully.');
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  #accessRequests = rxResource<AccessRequest[], string | null>({
```

Comment:
This should be called more specifically userAccessRequests, because it returns only the requests of the current user. We will also need a resource allAccessRequets that returns the requests of all users for the access request manager. I'm using this approach in the ivaService.

Currently the service loads the requests as soon as the user logs in. In the ivaService I added a method loadUserIvas that triggers the loading only upon explicit request.